src/hacbs/components/ApplicationDetails/tabs/overview/visualization/utils/visualization-utils.ts

```diff
@@ -44,8 +119,93 @@ export const getLatestResource = (resources = []) =>
         new Date(b.metadata.creationTimestamp).getTime() -
         new Date(a.metadata.creationTimestamp).getTime(),
     )
+    .slice(0, 1)
     .shift();
 
+export const workflowToNodes = (workflow: Workflow): WorkflowNode[] => {
+  const workflowDag = new DAG();
+
+  Object.keys(workflow).map((key) => {
+    const { id, data, runBefore, runAfter, isAbstractNode, runAfterResourceKey } = workflow[key];
+    const resources: WorkflowResources = data.resources || [];
+    const isParallelNode = !isAbstractNode && resources.length > 1;
+    const isDisabled = resources.length === 0;
+
+    if (isAbstractNode || isDisabled) {
+      const label = isDisabled ? `No ${data.label} set` : data.label;
+
+      const wData = {
+        ...data,
+        isDisabled,
+        isParallelNode,
+        resources,
+        label,
+        width: getLabelWidth(label),
+      };
+
+      workflowDag.addEdges(id, wData, runBefore, runAfter);
+    } else {
+      const rootResources = !runAfterResourceKey
+        ? resources
+        : (resources as []).filter((r) => {
+            return !get(r, runAfterResourceKey);
+          });
```

Comment:
`r` is of type `never` here.
Avoid casting `[]` by doing proper checks.

Reply:
Typescript throws an error when we use Union of array types  for methods like filter(), find(), every() and there is an [open issue](https://github.com/microsoft/TypeScript/issues/44373) on Typescript to provide a fix for this. :(

Reply:
Ah that's too bad :(
Ok work around is fine for now. Thanks!